### DISCUSSION TITLE
How to update Apollo Client to v3

### DISCUSSION THREAD

**Discussion by sve-odoo** [Akryum/vue-apollo#1079] [Q&A]
Hello,

My Vue app has a super standard setup with vue-cli-plugin-apollo.

It looks like Vue-apollo is compatible with Apollo Client 3. I would like to use it. How can I upgrade?

Thanks in advance!

**Comment by mmouterde** [comment]
I recently migrated to Apollo Client 3, It seems to work (at least for my use).

I simply had to :

- rewrite all my updateFn to use the new cache api (https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/#breaking-cache-changes)
- take care of immutable cache

**Comment by sve-odoo** [comment]
Thank you!
And to update the package, did you just update it using NPM?

**Comment by mmouterde** [comment]
Unfortunately not... to use Apollo Cient 3 you have to consider the imports updates too : https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/#updating-imports

**Comment by p3rh5g** [comment]
In TypeScript I get this error:

```
30:5 Type 'ApolloClient<NormalizedCacheObject>' is missing the following
properties from type 'ApolloClient<any>': store, writeData, initQueryManager
    28 | 
    29 | const apolloProvider = new VueApollo({
  > 30 |     defaultClient: apolloClient,
       |     ^
    31 | });
    32 | 
```

**Comment by quentinus95** [comment]
Asked the question here: https://github.com/vuejs/vue-apollo/issues/1059#issuecomment-717360611, cross posting in this issue.

cc @Akryum

**Comment by HansCronau** [comment]
> It looks like Vue-apollo is compatible with Apollo Client 3.

It is not. `vue-apollo` 3 is compatible with `apollo-client` 2. If you would like to move to `@apollo/client` 3, you can try out [Vue Apollo 4], which currently is in alpha.

[Vue Apollo 4]: https://v4.apollo.vuejs.org/guide/installation.html

**Reply by Akryum** [reply]
It is compatible except for the types. PR welcome!

**Reply by HansCronau** [reply]
True! If you're not using types, there shouldn't be a problem.
